### ISSUE TITLE
make.bat: autodetect GOROOT_BOOTSTRAP

### ISSUE THREAD

**Issue by FiloSottile** [golang/go#28641]
On Unix we automatically use `$(go env GOROOT)` as `GOROOT_BOOTSTRAP` (see #14339). We should do that on Windows as well, where it's harder to set env vars, and for consistency.

Make sure that there are checks in place to prevent `$GOROOT_BOOTSTRAP == $GOROOT`.

**Comment by crvv** [comment]
There is a CL for this.
https://go-review.googlesource.com/c/go/+/96455

**Comment by alexbrainman** [comment]
> On Unix we automatically use `$(go env GOROOT)` as `GOROOT_BOOTSTRAP` (see #14339). We should do that on Windows as well, where it's harder to set env vars, and for consistency.
> 
> Make sure that there are checks in place to prevent `$GOROOT_BOOTSTRAP == $GOROOT`.

@FiloSottile why change make.bat? Can we make all these changes in cmd/dist instead?

We had complicated make.bat before. But then we created cmd/dist and moved all complicated logic there. It is OK to do simple things, like setting an environment variable or calling a program, in make.bat. But what you are asking, is not simple.

This https://go-review.googlesource.com/c/go/+/208197/5 is the latest version of this fix. I did not verify it yet. And it is incomplete, because it will not compare GOROOT and GOROOT_BOOTSTRAP values properly - we need case-insensitive comparison here. Are you comfortable with CL 208197 changes? Will you be OK to debug / support this make.bat when it breaks? Because I am not comfortable.

And, if we change cmd/dist, it will cover all GOOS platforms with single code (mostly).

Alex
